packages/@aws-cdk/aws-autoscaling-hooktargets/lib/lambda-hook.ts

```diff
@@ -23,16 +24,22 @@ export class FunctionHook implements autoscaling.ILifecycleHookTarget {
   constructor(private readonly fn: lambda.IFunction, private readonly encryptionKey?: kms.IKey) {
   }
 
-  public bind(scope: Construct, lifecycleHook: autoscaling.ILifecycleHook): autoscaling.LifecycleHookTargetConfig {
-    const topic = new sns.Topic(scope, 'Topic', {
+  public bind(_scope: Construct, lifecycleHook: autoscaling.ILifecycleHook): autoscaling.LifecycleHookTargetConfig {
+    const topic = new sns.Topic(_scope, 'Topic', {
       masterKey: this.encryptionKey,
     });
+    try { lifecycleHook.role; } catch (noRoleError) {
```

Comment:
Ew :/

How about we change the signature of `bind()` to:

```ts
public bind(scope: Construct, options: BindHookTargetOptions): BindHookTargetResult;
```

Then we will have:

```
interface BindHookTargetOptions {
  readonly lifecycleHook: autoscaling.ILifecycleHook;
  readonly role?: iam.IRole;
}

interface BindHookTargetResult {
  readonly createdRole?: iam.IRole;
}
```

The linter will yell at you that it's a breaking change, but you have my permission to silence it (by adding it to `allowed-breaking-changes.txt`).